src/Internal/Service/OrderService.php

```diff
@@ -421,4 +422,16 @@ protected function get_order( int $order_id ): WC_Order {
 		}
 		return $order;
 	}
+
+	/**
+	 * Updates the '_wcpay_mode' meta data on an order.
+	 *
+	 * @param WC_Order $order The order.
+	 * @param string   $mode  Mode from the context.
+	 */
+	private function update_mode_meta_data( WC_Order $order, string $mode ) : void {
```

Comment:
To keep things consistent with the other methods, let's:

1. Make this method public.
2. Switch from using the `WC_Order` object to an order ID.
3. Add a getter for the meta, even though it would not be used for now.
4. Maybe rename this from `update_mode_meta_data` to `set_mode`? From outside the service, the mode should be only a property of the order, regardless of how it is stored.

Reply:
Because that is the way the service works. We don't want to carry `WC_Order` objects around, but work with IDs instead. Because of that, you'll notice how `_deprecated_get_order` is not "just" a method of the service, but is mostly used internally.

Reply:
>We don't want to carry WC_Order objects around, but work with IDs
Internally when using these private functions, isn't that an overhead, passing around IDs and constructing WC_Order again and again?

Reply:
We've already discussed this, and the conclusion is explained here: paJDYF-9hL-p2#comment-21012

If we see that there's additional overhead, we could introduce a small caching layer within the internal `get_order()` method, but the point is that the order service should be the link between WooPayments and Woo core, and nothing outside the service should be aware of `WC_Order` instances. 

> when using these private functions,

I do not see a reason to keep the mode setter+getter pair private.

Reply:
Thank you for explaining it.

Reply:
Changes made.